spec/system/tasks_spec.rb

```diff
@@ -0,0 +1,145 @@
+        end
+      end
+      context 'タスクの新規登録ページにアクセス' do
+        it '新規登録ページへのアクセスに失敗する' do
+          visit new_task_path
+          expect(current_path).to eq login_path
+          expect(page).to have_content('Login required')
+        end
+      end
+      context 'タスクの編集ページにアクセス' do
+        it '編集ページへのアクセスに失敗する' do
+          visit edit_task_path(task.id)
+          expect(current_path).to eq login_path
+          expect(page).to have_content('Login required')
+        end
+      end
+    end
+  end
+
+  describe 'ログイン後' do
+    before { login(user) }
+
+    describe 'タスク作成' do
+      context 'フォームの入力値が正常' do
+        it 'タスクの新規作成が成功する' do
+          visit new_task_path
+          fill_in 'Title', with: 'test_title'
+          fill_in 'Content', with: 'test_content'
+          select 'doing', from: 'Status'
+          fill_in 'Deadline', with: DateTime.new(2020, 6, 1, 10, 30)
+          click_button 'Create Task'
+          expect(page).to have_content 'Title: test_title'
+          expect(page).to have_content 'Content: test_content'
+          expect(page).to have_content 'Status: doing'
+          expect(page).to have_content 'Deadline: 2020/6/1 10:30'
+          expect(current_path).to eq '/tasks/1'
+        end
+      end
+      context 'タイトルが未入力' do
+        it 'タスク作成に失敗する' do
+          click_on 'New Task'
+          fill_in 'Title', with: ''
+          fill_in 'Content', with: 'content'
+          select 'todo', from: 'Status'
+          click_button 'Create Task'
+          expect(current_path).to eq tasks_path
+          expect(page).to have_content '1 error prohibited this task from being saved'
+          expect(page).to have_content("Title can't be blank")
+        end
+      end
+      context 'タイトルが重複' do
+        it 'タスク作成に失敗する' do
+          click_on 'New Task'
+          fill_in 'Title', with: task.title
+          fill_in 'Content', with: 'content'
+          select 'todo', from: 'Status'
+          click_button 'Create Task'
+          expect(current_path).to eq tasks_path
+          expect(page).to have_content '1 error prohibited this task from being saved'
+          expect(page).to have_content("Title has already been taken")
+        end
+      end
+    end
+
+    describe 'タスク編集' do
+      let!(:task) { create(:task, user: user) }
+      let(:other_task) { create(:task, user: user) }
+      before { visit edit_task_path(task) }
+
+      context 'フォームの入力値が正常' do
+        it 'タスク編集に成功する' do
+          fill_in 'Title', with: 'update_title'
+          fill_in 'Content', with: 'update_content'
+          select 'todo', from: 'Status'
+          click_button 'Update Task'
+          expect(current_path).to eq task_path(task)
+          expect(page).to have_content 'Title: update_title'
+          expect(page).to have_content 'Content: update_content'
+          expect(page).to have_content 'Status: todo'
+          expect(page).to have_content("Task was successfully updated.")
```

Comment:
ちなみに、このように, で区切ってカスタムエラーメッセージを定義することもできるので共有いたします。
```suggestion
          expect(page).to have_content("Task was successfully updated."), "タスク更新完了のメッセージが表示されていません。"
```

Reply:
共有ありがとうございます。

> 私は、エラーがたくさんでてわかりにくいときに、間違っている可能性がある場所を記述するのに使ってみています。

エラー内容を把握しやすくするためってことですね。
テスト量が増えると便利なのかなっと思いました。ありがとうございます！！

http://tkymtk.hatenablog.com/entry/rspec-customized-error-message-2013-12